syntax = "proto3";
import "data_point.proto";
import "bucketing_strategy.proto";
import "interpolate_strategy.proto";
import "l2_order_book.proto";
import "vectorization_strategy.proto";
import "stonk_aggregate.proto";
import "subreddit_top_listing.proto";
import "option_quote.proto";


message FeatureConfig {
  DataPointType data_point_type = 1;
  uint64 bucket_width_ms = 2;
  BucketingStrategy bucketing_strategy = 3;
  InterpolateStrategy interpolate_strategy = 4;
  VectorizationStrategy vectorization_strategy = 6;

  L2OrderBookConfig l2_order_book_config = 5;
  StonkAggregateConfig stonk_aggregate_config = 7;
  SubredditTopListingConfig subreddit_top_listing_config = 8;
  OptionQuoteConfig option_quote_config = 9;
}
